test/features/authentication_test.rb

```diff
@@ -0,0 +1,22 @@
+require './test/test_helper'
+
+class Authentication < ActionDispatch::IntegrationTest
+
+  def test_person_can_log_in_from_omniauth
+   person = Person.create(:provider => "github", :uid => "1", :first_name => "goldfish", :oauth_token => "token")
+   log_in(person)
+   save_and_open_page
+   assert page.has_content? "Your Jobs Dashboard"
+  end
+
+ def log_in(person)
+   OmniAuth.config.test_mode = true
+   OmniAuth.config.mock_auth[:github] = OmniAuth::AuthHash.new({
+     'provider' => 'github',
```

Comment:
Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.